lib/eventFactory.js

```diff
@@ -2,9 +2,111 @@
 
 const BaseFactory = require('./baseFactory');
 const Event = require('./event');
+const jobConfigSchema = require('screwdriver-data-schema').config.job;
+const joi = require('joi');
 
 let instance;
 
+/**
+ * Create a job
+ * @method createJob
+ * @param  {JobFactory} jobFactory  Job factory
+ * @param  {Job}        job         Job object
+ * @param  {String}     jobName     Job name
+ * @return {Job}                    Job object
+ */
+function createJob(jobFactory, job, jobName) {
+    return jobFactory.create({
+        pipelineId: job.pipelineId,
+        name: jobName,
+        permutations: job.permutations
+    });
+}
+
+/**
+ * Create builds associated with this event
+ * For example, if startFrom ~commit, then create builds with jobs that have requires: ~commit
+ * @method createBuilds
+ * @param  {Pipeline} pipeline                   Pipeline to create builds
+ * @param  {Object}   config
+ * @param  {String}   config.sha                 SHA this project was built on
+ * @param  {String}   config.username            Username of the user that creates this event
+ * @param  {String}   config.scmContext          The scm context to which user belongs
+ * @param  {String}   [config.prRef]             Ref if it's a PR event
+ * @param  {String}   [config.prNum]             PR Number if it's a PR event
+ * @param  {String}   [config.startFrom]         Where the event starts from (jobname or ~commit, ~pr, etc)
+ *                                               Making optional for backward compatibility
+ * @param  {Number}   eventId                    Event's id
+ */
+function createBuilds(pipeline, config, eventId) {
+    const startFrom = config.startFrom;
+
+    // If no startFrom, then do nothing. Remove once we switch to new workflow design
+    if (!startFrom) {
+        return null;
+    }
+
+    /* eslint-disable global-require */
+    const BuildFactory = require('./buildFactory');
+    const buildFactory = BuildFactory.getInstance();
+    const JobFactory = require('./jobFactory');
+    const jobFactory = JobFactory.getInstance();
+    /* eslint-enable global-require */
+
+    return pipeline.jobs.then((jobs) => {
```

Comment:
How does this behave when there are 2+ PR's open at the same time ? Does `pipeline.jobs` return all ? If so we need to filter out jobs with Pr Number not matching config.prNum.

I think we need logic like ``` if (config.startFrom === '~pr' && job.isPR() && job.prNum !== config.prNum) return ```

Reply:
Made changes to handle this case